app/helpers/task_list_helper.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+module TaskListHelper
+  def row_helper(trainee, task)
+    task_list = {
+      publish_course_details:
+        {
+          task_name: "Course details",
+          path: edit_trainee_publish_course_details_path(trainee),
+          confirm_path: trainee_course_details_confirm_path(trainee),
+          classes: "course-details",
+          status: ProgressService.call(
+            validator: ValidatePublishCourseForm.new(trainee),
+            progress_value: trainee.progress.course_details,
+          ).status,
+        },
+
+      course_details:
+        {
+          task_name: "Course details",
+          path: edit_trainee_course_details_path(trainee),
+          confirm_path: trainee_course_details_confirm_path(trainee),
+          classes: "course-details",
+          status: ProgressService.call(
+            validator: CourseDetailsForm.new(trainee),
+            progress_value: trainee.progress.course_details,
+          ).status,
+        },
+
+      training_details:
+        {
+          task_name: (trainee.apply_application? ? "Training details" : "Trainee start date and ID"),
+          path: edit_trainee_training_details_path(trainee),
+          confirm_path: trainee_training_details_confirm_path(trainee),
+          classes: "training-details",
+          hint_text: "Start date and trainee ID",
+          status: ProgressService.call(
+            validator: TrainingDetailsForm.new(trainee),
+            progress_value: trainee.progress.training_details,
+          ).status,
+        },
+
+      placement_details:
+        {
+          task_name: "Placement details",
+          path: review_draft_trainee_path(trainee),
+          confirm_path: review_draft_trainee_path(trainee),
+          classes: "placement-details",
+          status: ProgressService.call(
+            validator: PlacementDetailForm.new(trainee),
+            progress_value: trainee.progress.placement_details,
+          ).status,
+        },
+
+      school_details:
+        {
+          task_name: "Lead and employing schools",
+          path: edit_trainee_lead_schools_path(trainee),
+          confirm_path: edit_trainee_lead_schools_path(trainee),
+          classes: "school-details",
+          status: ProgressService.call(
+            validator: Schools::FormValidator.new(trainee),
+            progress_value: trainee.progress.schools,
+          ).status,
+        },
+
+      trainee_data:
+        {
+          task_name: "Trainee data",
+          path: "#",
+          confirm_path: "#",
```

Comment:
The trainee data page has been added now https://github.com/DFE-Digital/register-trainee-teachers/pull/864 so we can add the link.

Reply:
I've added the path, though I'm not sure how the `progress_value` would work here as it's a combination of different sections being completed, probs beyond scope of ticket?